internal/gatewayapi/securitypolicy.go

```diff
@@ -398,6 +398,26 @@ func resolveSecurityPolicyGatewayTargetRef(
 		return nil, nil
 	}
 
+	// If sectionName is set, make sure its valid
+	if target.SectionName != nil {
```

Comment:
this section name checking logic is same as in client-traffic-policy, is there any chance we can wrap them up in a common function ?